modPrecompiled/test/org/aion/precompiled/Blake2bHashTest.java

```diff
@@ -0,0 +1,161 @@
+ *     be useful, but WITHOUT ANY WARRANTY; without even the implied
+ *     warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *     See the GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with the aion network project source files.
+ *     If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Contributors:
+ *     Aion foundation.
+ */
+package org.aion.precompiled;
+
+import static com.google.common.truth.Truth.assertThat;
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+
+import java.nio.charset.StandardCharsets;
+import org.aion.base.util.ByteUtil;
+import org.aion.crypto.hash.Blake2b;
+import org.aion.precompiled.contracts.Blake2bHashContract;
+import org.aion.vm.AbstractExecutionResult.ResultCode;
+import org.aion.vm.ExecutionResult;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class Blake2bHashTest {
+
+    private static final long INPUT_NRG = 1000;
+
+    private byte[] byteArray1 = "a0010101010101010101010101".getBytes();
+    private byte[] byteArray2 = "1".getBytes();
+    private byte[] shortByteArray = "".getBytes();
+    private byte[] bigByteArray = new byte[2*1024*1024];
+    private byte[] bigByteArray2 = new byte[2*1024*1024+1];
+    private Blake2bHashContract blake2bHasher;
+
+    @Before
+    public void setUp() {
+        blake2bHasher = new Blake2bHashContract();
+    }
+
+    @Test
+    public void testBlake256() {
+        ExecutionResult res = blake2bHasher.execute(byteArray1, INPUT_NRG);
+        byte[] output = res.getOutput();
+
+        assertEquals(ResultCode.SUCCESS, res.getResultCode());
+        assertEquals(32, output.length);
+        String blake2bStr1 = "aa6648de0988479263cf3730a48ef744d238b96a5954aa77d647ae965d3f7715";
+        assertEquals(blake2bStr1, ByteUtil.toHexString(output));
+    }
+
+    @Test
+    public void testBlake256_2() {
+        ExecutionResult res = blake2bHasher.execute(byteArray2, INPUT_NRG);
+        byte[] output = res.getOutput();
+
+        assertEquals(ResultCode.SUCCESS, res.getResultCode());
+        assertEquals(32, output.length);
+
+        String blake2bStr2 = "92cdf578c47085a5992256f0dcf97d0b19f1f1c9de4d5fe30c3ace6191b6e5db";
+        assertEquals(blake2bStr2, ByteUtil.toHexString(output));
+    }
+
+    @Test
+    public void testBlake256_3() {
+        ExecutionResult res = blake2bHasher.execute(bigByteArray, 2_000_000L);
+        byte[] output = res.getOutput();
+
+        assertEquals(ResultCode.SUCCESS, res.getResultCode());
+        assertEquals(32, output.length);
+
+        String blake2bStr2 = "9852d74e002f23d14ba2638b905609419bd16e50843ac147ccf4d509ed2c9dfc";
+        assertEquals(blake2bStr2, ByteUtil.toHexString(output));
+    }
+
+    @Test
+    @Ignore
+    public void testBlake128() {
```

Comment:
I noticed we deleted the blake128 method ... should we just delete this test as well?

Reply:
sure